guides/BACKEND.md

```diff
@@ -0,0 +1,226 @@
+
+    - **Example 2:** Table Name: "map" && Table ID: "map_id" (as a UUID)
+
+2. You should be mapping objects as foreign keys to their respective UUID. For example if you made a new table that holds a player's messages, it would be something like:
+
+    - From: player_id
+    
+    - To: Player.id
+
+### Updating the GraphQL schemas
+
+Assuming the Docker Containers are running. After creating the table. You will need to run:
+
+```bash
+yarn codegen update-schema
+
+# and then
+yarn codegen generate
+```
+
+This will update the `.graphql` schemas for Hasura and make it easier to create graphql queries for the frontend.
+
+#### Permissions
+
+By default, only admins are allowed to change the permissions. In order to query, add and update data as other than the admin, you'll need to change the permissions.
+
+1. You will need to select the new table you created on the data the Data tab in the Hasura Console.
+
+2. You will need to select the table you created under `Tables/Views/Functions`.
+
+3. Select the permissions tab and then select either `insert`, `select`, `update` and `delete` for `player` or `public`.
+
+4. Generally, you will want to have a custom check. For the most part, data should only be inserted and deleted based on their user id. The following is a custom check you could create for `Row insert permissions` or `Pre-update check` or `Row delete permissions`.
+
+```json
+{
+    "id": {
+        "_eq":"X-Hasura-User-Id"
+    }
+}
+```
+
+5. Furthermore, when selecting or updating data. You can add permissions for specific columns. You select which ones should be allowed via the provided checkboxes.
+
+6. Finally, make sure that the changes for permissions are updated in `hasura/metadata/tables.yaml`.
+`Pre-update check`nsert and read data immediately. There are pre-generated functions that come with Hasura. For creating new entries. The following are example queries you could send immediately to `http://localhost:8080/v1/graphql`.
+
+
+1. Insert a new item
+
+```graphql
+# Insert a new item, assuming `Item` is the table name
+mutation insert {
+  # Built in function with Hasura GraphQL
+  insert_Item_one(
+    # The respective columns
+    object: {
+      data: "..."
+    }
+  ) {
+    # The keys to return on a successful insertion
+    id
+    data
+  }
+}
+```
+
+As long as it fits the constraints of the table (ie: Foreign Key uniqueness, Data types), the insertion should be successful.
+
+2. Update an item by id
+
+```graphql
+mutation update {
+    # Built in Hasura function
+    update_Item(
+        where: {
+            # Can also use _gt, _gte, _lt, _lte, _neq etc.
+            # Can also specify any column
+            id: {_eq: "[UUID]"}
+        },
+        # The columns you want to update
+        _set: {
+            data: "..."
+        }) {
+        # you can either supply `returning` or `affected_rows` for the response
+        affected_rows
+    }
+}
```

Comment:
In the case where you're updating single items by their id / primary key, a better way to do it is use `update_Item_by_pk`.